### DISCUSSION TITLE
Controlling dropdown widths

### DISCUSSION THREAD

**Discussion by kevinbeal** [mantinedev/mantine#1226] [Q&A] [answered]
Hello! Nice components you got here. Thanks for making them free.

I was wondering how I can make dropdowns from `<Select />` and `<Autocomplete />` not constrain their widths to the width of the inputs they control.

For example, I have this dropdown: 
![image](https://user-images.githubusercontent.com/855252/162766234-0612ed25-5f8e-41c7-8b69-303c695e6082.png)

It shows a horizontal scroll bar when you hover over it. It would be nice if there were a `scrollAreaProps` or `paperProps` prop (or whatever is relevant) on the `<Select />`/`<Autocomplete />` components and something there that would allow it to grow in size, like `{ style: { right: "auto" } }`. But I don't know how it actually works. I mean only to convey my intention.

Any guidance would be appreciated. Thank you :)

**Comment by rtivital** [comment]
Select and Autocomplete dropdown width cannot be changed, if you need this feature, use Popover component instead.

**Reply by escobar5** [reply]
Is there any reason for this? it would be useful to have a Select with a Dropdown that has a different width, the one in that picture is conceptually a Select, not a Popover

**Comment by avigoldman** [comment]
<img width="417" alt="Screen Shot 2022-11-29 at 9 55 40 AM" src="https://user-images.githubusercontent.com/4924860/204562669-ec4e35a5-046d-4334-b84b-853d91075dcc.png">

I was able to achieve this on the `<Select />` with:

```
<Box pos="relative">
      <Select
        w={256}
        sx={{
          '.mantine-Select-dropdown': {
            width: '400px !important',
            left: '0 !important',
          },
        }}
      />
    </Box>
```

The `<Box/>` with position=relative, keeps forces the dropdown to be aligned left with the `<Select />`

**Reply by arheinjohnson** [reply]
From what I can tell this is from an older version of mantine; the sx prop is no longer available

**Comment by bozosort** [comment]
You can change the width through Styles API by overriding width of wrapper element of Autocomplete. You just need to add one line of code:
            <Autocomplete
              **styles={{ wrapper: { width: 600 } }}**
              placeholder="Pick one"
              data={['React', 'Angular', 'Svelte', 'Vue']}
            />

You can find all inner elements of a component in styles tab of mantine documentation:(https://mantine.dev/core/autocomplete/?t=styles-api)
For more details on styles API, you can check:(https://mantine.dev/styles/styles-api/)

**Reply by petrvecera** [reply]
wow thanks a lot, you saved me! 

I am baffled that you can't setup this directly on the select component , when you have big ass selector for 2 characters options 😮‍💨

**Reply by sovetski** [reply]
Thank you, you saved my day!!

**Comment by AlonMiz** [comment]
it would be beneficial to have the dropdown width adjusted depending on the width of the options.
and not fixed by the width of the Select.
similar to what ReactSelect does

**Reply by sovetski** [reply]
+1

**Reply by MaximBuz** [reply]
+1

**Reply by gemueselasagne** [reply]
+1

**Reply by eliobricenov** [reply]
+1

**Reply by pmprete** [reply]
+1

**Reply by ojamui** [reply]
+1

**Reply by Marjona6** [reply]
+1, would really love to see this.

**Reply by AndrejNemec** [reply]
+1

**Reply by arheinjohnson** [reply]
+1

**Reply by laugri** [reply]
Hey @rtivital, what is your stance on this ?

**Reply by rtivital** [reply]
You can do that with `comboboxProps={{ width: 200 }}`

**Comment by rtivital** [comment]
The feature is available with `comboboxProps={{ width: 200 }}`

**Comment by daneverso** [comment]
In 7.16.1:

```
.selectInput {
  width: 250px !important;
}

```
and after importing classes:

```
<Select
  classNames={{
    input: classes.selectInput
   }}
 />
```

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/core/select/#dropdown-width
